chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -40,13 +40,25 @@
  */
 public class Octree {
```

Comment:
Maybe change `Octree` to be the interface that `OctreeImplementation` is at the moment, let `NodeBasedOctree` and `PackedOctree` implement `Octree` and create a `OctreeFactory` class that creates instances of the octree implementation the user wants?

Not sure about this, but seems to be a more Java-like way. Also, there wouldn't be that `implementation` field and manual "redirects" to the actual implementation in the method calls needed. :thinking:

Reply:
Well, as you said next. Keeping it like that, albeit not really idiomatic, let us swap the implementation without having to update any reference to the tree